poll/poll_func.go

```diff
@@ -2,75 +2,103 @@ package poll
 
 import (
 	"fmt"
+	"github.com/mattermost/platform/model"
+	"io"
 	"log"
 	"net/http"
+	"time"
+)
 
-	"github.com/mattermost/platform/model"
+const (
+	RESPONSE_USERNAME = "Matterpoll"
+	RESPONSE_ICON_URL = "https://www.mattermost.org/wp-content/uploads/2016/04/icon.png"
 )
 
 var Conf *PollConf
 
 func PollCmd(w http.ResponseWriter, r *http.Request) {
 	defer r.Body.Close()
-	err := r.ParseForm()
-	if err != nil {
-		log.Print(err)
+	// Check if Content Type is correct
+	if r.Header.Get("Content-Type") != "application/x-www-form-urlencoded" {
+		w.WriteHeader(http.StatusUnsupportedMediaType)
 		return
 	}
-	poll, err := NewPollRequest(r.Form)
+	err := r.ParseForm()
 	if err != nil {
-		log.Print(err)
-		return
-	}
-	if len(Conf.Token) != 0 && Conf.Token != poll.Token {
-		log.Print("Token missmatch. Check you config.json")
+		w.WriteHeader(http.StatusBadRequest)
 		return
 	}
+	poll, err := NewPollRequest(r.Form)
+	valid_poll := err == nil
 
-	c := model.NewClient(Conf.Host)
-	c.TeamId = poll.TeamId
-
-	_, err = login(c)
-	if err != nil {
-		log.Print(err)
-		return
+	var response model.CommandResponse
+	response.Username = RESPONSE_USERNAME
+	response.IconURL = RESPONSE_ICON_URL
+	if valid_poll {
+		response.ResponseType = model.COMMAND_RESPONSE_TYPE_IN_CHANNEL
+		response.Text = poll.Message + ` #poll`
+	} else {
+		response.ResponseType = model.COMMAND_RESPONSE_TYPE_EPHEMERAL
+		response.Text = err.Error()
 	}
-	p, err := post(c, poll)
-	if err != nil {
-		log.Print(err)
-		return
+	io.WriteString(w, response.ToJson())
+	if valid_poll {
+		if len(Conf.Token) != 0 && Conf.Token != poll.Token {
+			log.Print("Token missmatch. Check you config.json")
+			return
```

Comment:
I suppose that we should inform users about invalid matterpoll settings.
Only by the looking a poll message with no reaction without error message, It's hard for user to understand what happens.

If users see the message about invalid server settings, user maybe ask admins confirming server settings.